app/src/main/java/com/infomaniak/drive/ui/fileList/preview/PreviewPDFFragment.kt

```diff
@@ -38,20 +39,32 @@ import com.infomaniak.drive.utils.IOFile
 import com.infomaniak.drive.utils.PreviewPDFUtils
 import com.infomaniak.lib.core.models.ApiResponse
 import com.infomaniak.lib.core.utils.safeBinding
+import com.infomaniak.lib.pdfview.listener.OnErrorListener
 import com.infomaniak.lib.pdfview.scroll.DefaultScrollHandle
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.Job
 import kotlinx.coroutines.launch
 
 class PreviewPDFFragment : PreviewFragment() {
+    private val previewPDFViewModel by viewModels<PreviewPDFViewModel>()
 
-    private var binding: FragmentPreviewPdfBinding by safeBinding()
+    private val navController by lazy { findNavController() }
 
-    private val previewPDFViewModel by viewModels<PreviewPDFViewModel>()
+    private var binding: FragmentPreviewPdfBinding by safeBinding()
 
     private var pdfFile: IOFile? = null
     private var isDownloading = false
-
+    private var hasEnteredPassword = false
+
+    private val onPdfLoadError: OnErrorListener = OnErrorListener {
+        navController.apply {
+            if (currentDestination?.id == R.id.pdfPasswordDialog) navigateUp()
```

Comment:
I'm not really sure you should use navigateUp here, is there a special use ? 
We always use popBackStack(), what do you think about it ?